src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -42,6 +100,84 @@ public static void PopulateDefaultPayload(IEnumerable<string> disabledFields, IH
             }
         }
 
+        /// <summary>
+        /// Gathers operational data from an Azure VM, if the IMDS endpoint is present. If not, and the SDK is
+        /// not running on an Azure VM with the Azure Instance Metadata Service running, the corresponding
+        /// fields will not get set into the heartbeat payload.
+        /// </summary>
+        private static async Task AddAzureVmDetail(HeartbeatProvider heartbeatManager, IEnumerable<string> enabledFields)
```

Comment:
There should be a configuration switch to disable any attempts to run this discovery

Reply:
Agreed, done.